x/posts/types/models/post.go

```diff
@@ -122,6 +124,15 @@ func NewPost(id, parentID PostID, message string, allowsComments bool, subspace
 		OptionalData:   optionalData,
 		Creator:        creator,
 	}
+
+	// PostID calculation
+	bz := []byte(parentID.String() + message + subspace + strconv.FormatBool(allowsComments) + created.String() +
+		creator.String())
+	hash := sha256.Sum256(bz)
+
+	post.PostID = PostID(hex.EncodeToString(hash[:]))
```

Comment:
I really like how cleanly this has been implemented :100: However, there is only one concern that comes to my mind: incompatibility with polls and attachments.

Let me explain this with an example. 

Let's suppose that Alice created three different posts and syncs them together, resulting in them being inserted in the same block. The posts are formed as follows 

1. First post  
   ```yml
   parent_id: "1"
   message: "Hello!"
   subspace: "mooncake"
   allows_comments: true
   ```

2. Second post
   ```yml
   parent_id: "1"
   message: "Hello!"
   subspace: "mooncake"
   allows_comments: true
   poll: 
     question: "Do you like cats?"
     options: 
        - "Yes"
        - "No"
   ```

3. Third post
   ```yml
   parent_id: "1"
   message: "Hello!"
   subspace: "mooncake"
   allows_comments: true
   attachments: 
     - url: "https://test.example.com/file" 
   ```

Now, all of them would have the same ID since this is computed **not** considering the poll nor the attachments info. 

In order to solve this problem, what I would suggest to do is the following. First, create a new private method: 

```go
// Computes a post ID based on the content of the given post.
func computePostId(post Post) string {
  // Hash together all the post data: 
  // ParentID, Message, Subspace, AllowsComments, Created, Creator
  // and also PollData and Attachments if not empty
}
```

Then, call this method in three different places: 

1. When the post is created the first time, inside the `NewPost` method. 
2. When some attachments are added to the post, inside `WithAttachments` 
3. When some poll data is added to the post, inside the `WithPollData` method

This way we would have a deterministic ID that is computed based on the whole post data, instead just part of it. Also, the post id would always be refreshed when some data is added/removed or changes.

Reply:
Perfect I didn't think about this before! I will provide to implment it asap